plans/network/pingpong.go

```diff
@@ -187,6 +206,96 @@ func pingpong(runenv *runtime.RunEnv, initCtx *run.InitContext) error {
 	return nil
 }
 
+// Exchanges own data network address with other instance in the test, and returns their data addres
+func exchangeListenAddrs(ctx context.Context, runenv *runtime.RunEnv, client sync.Client) (string, error) {
+
+	// Retrieve own data IP
+	tcpAddr, err := getSubnetAddr(runenv.TestSubnet)
+	if err != nil {
+		return "", err
+	}
+
+	mynode := &ListenAddrs{}
+	mine := map[string]struct{}{}
+
+	mynode.Addrs = append(mynode.Addrs, tcpAddr.String())
+	for _, l := range mynode.Addrs {
+		mine[l] = struct{}{}
+	}
+
+	_ = client.MustSignalAndWait(ctx, sync.State("listening"), runenv.TestInstanceCount)
+
+	allAddrs, err := shareAddresses(ctx, client, runenv, mynode)
+	if err != nil {
+		return "", err
+	}
+
+	otherAddrs := []string{}
+	// filter addresses to remove any own address
+	for _, addr := range allAddrs {
+		if _, ok := mine[addr]; ok {
+			continue
+		}
+		otherAddrs = append(otherAddrs, addr)
+	}
+
+	_ = client.MustSignalAndWait(ctx, sync.State("got-other-addrs"), runenv.TestInstanceCount)
+
+	// since there is only 1 other instance in test, we expect a single address in the list
+	if len(otherAddrs) > 1 {
+		return "", fmt.Errorf("expected a single address to be returned, got %d", len(otherAddrs))
+	}
+
+	return otherAddrs[0], nil
```

Comment:
It looks like this function assumes a single `tcpAddr` from `getSubnetAddr` and at the same time acts like we're moving around arrays of addresses.

Could we be explicit about exchanging a single address? That would simplify the code a lot.

Reply:
That's exactly my thinking - https://github.com/testground/testground/pull/1462#discussion_r976896605

Reply:
(resolving in favor of your comment @galargh, thank you o7)